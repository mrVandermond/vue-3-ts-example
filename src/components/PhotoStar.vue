<template>
  <div class="photo-star">
    <img
      :src="starUrl"
      alt="star-image"
    >
  </div>
</template>

<script lang="ts" setup>
import { computed } from 'vue';

const props = defineProps<{
  isActive: boolean;
}>();
const starUrl = computed((): string => {
  if (props.isActive) {
    return 'assets/star_active.png';
  }

  return 'assets/star_empty.png';
});
</script>

<style lang="scss">
.photo-star {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
