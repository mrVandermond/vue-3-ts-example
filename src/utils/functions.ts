export function getItemIndexBy<T>(array: T[], item: T, field: keyof T): number {
  return array.findIndex((temp) => temp[field] === item[field]);
}
