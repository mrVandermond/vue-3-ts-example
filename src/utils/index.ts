export * as utlFuncs from './functions';
export * as utlCst from './constants';
