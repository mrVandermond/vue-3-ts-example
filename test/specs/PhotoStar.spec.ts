import PhotoStar from '@/components/PhotoStar.vue';

import type { TWrapperFactoryReturnType } from '../utils';
import { wrapperFactory } from '../utils';

describe('PhotoStar', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const getWrapper = (props: any): TWrapperFactoryReturnType => wrapperFactory(PhotoStar, {
    props,
  });

  it('Рендер в активном состоянии', () => {
    const wrapper = getWrapper({
      isActive: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Рендер в неактивном состоянии', () => {
    const wrapper = getWrapper({
      isActive: false,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
