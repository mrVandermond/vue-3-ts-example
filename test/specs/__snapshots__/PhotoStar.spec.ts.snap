// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PhotoStar > Рендер в активном состоянии 1`] = `"<div class=\\"photo-star\\"><img src=\\"assets/star_active.png\\" alt=\\"star-image\\"></div>"`;

exports[`PhotoStar > Рендер в неактивном состоянии 1`] = `"<div class=\\"photo-star\\"><img src=\\"assets/star_empty.png\\" alt=\\"star-image\\"></div>"`;
